### PR TITLE
Azure Digital Twins 2020-05-31-preview

### PR DESCRIPTION
**Pull request by briancr-ms**
Add the Azure Digital Twins 2020-05-31-preview data plane OpenAPI spec.

### CODE REVIEW

**Comment by akning-ms** [comment]
Looks host should not have special charater like <>. @raych1 for comments if have

**Comment by raych1** [comment]
Hi @briancr-ms , domain name must start with a letter, end with a letter or digit and have as interior characters only letter, digits, and hyphen according to the [RFC spec](https://tools.ietf.org/html/rfc1035). If you want to use host template, then try this extension [x-ms-parameterized-host](http://azure.github.io/autorest/extensions/#x-ms-parameterized-host).

**Comment by adrianhall** [comment]
Azure REST API Review Board Result:
- All REQUIRED issues taken
- RECOMMENDATIONS will be adopted for GA release.

Unless the reviewer finds something else that needs immediate action, no blockers remain.

File: specification/digitaltwins/data-plane/readme.go.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+```
+
+### Tag: package-2020-05-31-preview and go
+
+These settings apply only when `--tag=package-2020-05-31-preview --go` is specified on the command line.
+Please also specify `--go-sdks-folder=<path to the root directory of your azure-sdk-for-go clone>`.
+
+```yaml $(tag) == 'package-2020-05-31-preview' && $(go)
+namespace: Azure.DigitalTwins.Core
+output-folder: $(go-sdks-folder)/digitaltwins/Generated
```
Comment (jhendrixMSFT) [inline specification/digitaltwins/data-plane/readme.go.md:18]: `output-folder: $(go-sdk-folder)/services/preview/$(namespace)/2020-05-31/$(namespace)`

File: specification/digitaltwins/data-plane/readme.go.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  clear-output-folder: true
+```
+
+### Tag: package-2020-05-31-preview and go
+
+These settings apply only when `--tag=package-2020-05-31-preview --go` is specified on the command line.
+Please also specify `--go-sdks-folder=<path to the root directory of your azure-sdk-for-go clone>`.
+
+```yaml $(tag) == 'package-2020-05-31-preview' && $(go)
+namespace: Azure.DigitalTwins.Core
```
Comment (jhendrixMSFT) [inline specification/digitaltwins/data-plane/readme.go.md:17]: `namespace: digitaltwins`

Please remove it from this block and add to the above block after `license-header`.

File: specification/digitaltwins/data-plane/Microsoft.DigitalTwins/preview/2020-05-31-preview/digitaltwins.json [hunk 3]
```diff
@@ -0,0 +1,1497 @@
...
+            },
+            "headers": {
+              "ETag": {
+                "description": "Weak Etag.",
+                "type": "string"
+              }
+            }
+          },
+          "202": {
+            "description": "Asynchronous Success"
```
Comment (jhendrixMSFT) [inline specification/digitaltwins/data-plane/Microsoft.DigitalTwins/preview/2020-05-31-preview/digitaltwins.json:382]: Does this mean it's an async operation (i.e. long running operation)?  If so then this operation should be annotated with `x-ms-long-running-operation`.  Please see https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-long-running-operation for details.

File: specification/digitaltwins/data-plane/Microsoft.DigitalTwins/preview/2020-05-31-preview/digitaltwins.json [hunk 4]
```diff
@@ -0,0 +1,1497 @@
...
+            "description": "Success",
+            "headers": {
+              "ETag": {
+                "description": "Weak Etag.",
+                "type": "string"
+              }
+            }
+          },
+          "202": {
+            "description": "Asynchronous Success"
```
Comment (jhendrixMSFT) [inline specification/digitaltwins/data-plane/Microsoft.DigitalTwins/preview/2020-05-31-preview/digitaltwins.json:477]: LRO?

File: specification/digitaltwins/data-plane/Microsoft.DigitalTwins/preview/2020-05-31-preview/digitaltwins.json [hunk 5]
```diff
@@ -0,0 +1,1497 @@
...
```
Comment (jhendrixMSFT) [inline specification/digitaltwins/data-plane/Microsoft.DigitalTwins/preview/2020-05-31-preview/digitaltwins.json:990]: LRO?

File: specification/digitaltwins/data-plane/Microsoft.DigitalTwins/preview/2020-05-31-preview/digitaltwins.json (hunk 3 above)
Comment (briancr-ms) [reply specification/digitaltwins/data-plane/Microsoft.DigitalTwins/preview/2020-05-31-preview/digitaltwins.json:382]: No, it's not a LRO in the x-ms-long-running-operation sense. This API can return 200 when the digital twin entity is created/updated immediately or 202 when the entity is created/updated and requires a device sync. In that case, the client uses the digital twin notification feed to get notifications about entity lifecycle and state changes.

File: specification/digitaltwins/data-plane/Microsoft.DigitalTwins/preview/2020-05-31-preview/digitaltwins.json (hunk 4 above)
Comment (briancr-ms) [reply specification/digitaltwins/data-plane/Microsoft.DigitalTwins/preview/2020-05-31-preview/digitaltwins.json:477]: Similar to the above comment. In the case when the digital twin represents a device that may or may not be connected, this API returns 202 and the client uses the digital twin notification feed to get updates.

File: specification/digitaltwins/data-plane/Microsoft.DigitalTwins/preview/2020-05-31-preview/digitaltwins.json (hunk 5 above)
Comment (briancr-ms) [reply specification/digitaltwins/data-plane/Microsoft.DigitalTwins/preview/2020-05-31-preview/digitaltwins.json:990]: Similar to the above comment. In the case when the digital twin represents a device that may or may not be connected, this API returns 202 and the client uses the digital twin notification feed to get updates.

**Review by jhendrixMSFT — APPROVED** [review]

**Comment by jhendrixMSFT** [comment]
@briancr-ms is this API live?

**Comment by briancr-ms** [comment]
This API will be live next week with the public preview release of Azure Digital Twins.

---
In reply to: [638530202](https://github.com/Azure/azure-rest-api-specs/pull/9622#issuecomment-638530202) [](ancestors = 638530202)

**Pull request merged**